### PR TITLE
Refactor test_image to plot @circuit.png instead of test_logo.png

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

So that we don't need to pull test_logo.png from dvc, or run test_logo first to generate the image.

The circuit.png file is from https://github.com/GenericMappingTools/gmtserver-admin/blob/master/cache/circuit.png

Fixes https://github.com/GenericMappingTools/pygmt/pull/2911#issuecomment-1869905218, patches https://github.com/GenericMappingTools/pygmt/pull/2079#discussion_r956734196, fixes https://github.com/GenericMappingTools/pygmt/pull/518#issuecomment-660431849

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]
OK to merge if all tests pass.

File: pygmt/tests/test_image.py
Comment (weiji14) [inline pygmt/tests/test_image.py:10]: Gonna try removing this line and see if it still crashes. Xref https://github.com/GenericMappingTools/pygmt/pull/518#issuecomment-657896532.

**Comment by weiji14** [comment]
Marking as 'ready for review' to run tests on Windows. Still need to revert the uncommented `pull_request:` lines before merging.

File: pygmt/tests/test_image.py
Comment (weiji14) [reply pygmt/tests/test_image.py]: Oo, `test_image` no longer crashes on Windows with GMT 6.3 (see https://github.com/GenericMappingTools/pygmt/actions/runs/7334833328/job/19971998592?pr=2922#step:7:508) and GMT 6.4 (see https://github.com/GenericMappingTools/pygmt/actions/runs/7334833574/job/19971999909?pr=2922#step:8:473) :tada:

**Comment by weiji14** [comment]
Do we still want to wait for ghostscript 10.02.1 first @seisman? Xref #2694.

**Comment by seisman** [comment]
> Do we still want to wait for ghostscript 10.02.1 first @seisman? Xref #2694.

I think we can merge this PR now, so that you can continue with benchmarks.

**Pull request merged**
